packages/slate-react/src/plugin/react-editor.ts

```diff
@@ -81,6 +83,29 @@ export const ReactEditor = {
     )
   },
 
+  /**
+   * Find the DOM node that implements DocumentOrShadowRoot for the editor.
+   */
+
+  findDocumentOrShadowRoot(editor: ReactEditor): Document | ShadowRoot {
+    const el = ReactEditor.toDOMNode(editor, editor)
+    const root = el.getRootNode()
+
+    if (!(root instanceof Document || root instanceof ShadowRoot))
+      throw new Error(
+        `Unable to find DocumentOrShadowRoot for editor element: ${el}`
```

Comment:
Noticed an interesting issue using this branch myself: The `selectionchange` event fires as the editor unmounts (we have a drawer component that contains the editor -- closing the drawer causes `selectionchange` to fire). When this happens, the `selectionchange` event fires and `findDocumentOrShadowRoot` is called. This error is hit like so:
```
// ... selectionchange event calls this function ...
const el = ReactEditor.toDOMNode(editor, editor) // el is the editor, which isn't currently in the DOM
const root = el.getRootNode() // root is another element that is not in the DOM or the shadow root
// error throws because root is not Document or ShadowRoot
```

Any ideas? No worries if not -- this could definitely be an issue that's unrelated to Slate.

Reply:
I couldn't reproduce this myself, yet. We are also unmounting our slate editor when the user navigates to a different section/page but so far I didn't get this error. 
Could you setup a minimal example illustrating this issue?
One idea how to get around this would be to call `blur()` before the drawer is closed but obviously this would just be a temporally workaround.

Reply:
FWIW, I'm seeing this same behavior with an unmounting editor... I tried `blur`ing it with a `useEffect` but that's probably too late.. still errors, but this time in the `ReactEditor.blur(editor)`.

Reply:
I'm trying to fix an issue related to this change, can you explain why we need to throw an exception here @davidruisinger? I believe I'm missing context. For reference here's my [draft pr](https://github.com/ianstormtaylor/slate/pull/4219) for fixing the iframe crash, would be good to get some 👀 on it before I add cypress tests

Reply:
At the very least, these `instanceof` checks should use `getWindow` the way other similar checks do

https://github.com/ianstormtaylor/slate/blob/6a1376332bbd2567336c444c57c1e64fdf706feb/packages/slate-react/src/components/editable.tsx#L383-L384

https://github.com/ianstormtaylor/slate/blob/8a2a60b052dd666ef4ec128863a56b251aec8ea8/packages/slate-react/src/utils/dom.ts#L67-L68